### ISSUE TITLE
Ranges can only match single char

### ISSUE THREAD

**Issue by xubuntu4iran** [koalaman/shellcheck#3140]
```
~/Documents/Math/shellcheck-v0.10.0/shellcheck rep-factors.sh 

In rep-factors.sh line 41:
comp=$(sed -n "${j}p" "${filename}" |  jq -r  .[\"input-decimal\"]);
                                               ^-----------------^ SC2102 (info): Ranges can only match single chars (mentioned due to duplicates).

For more information:
  https://www.shellcheck.net/wiki/SC2102 -- Ranges can only match single char...
```
1.10 stable release used

**Comment by brother** [comment]
I understand what you are reaching towards but this is frankly speaking a pretty lazy and bad issue description. 

Ignore the problem with a directive.

**Comment by sideeffect42** [comment]
Please do not ignore this shellcheck message. It is completely valid and refers to a bug in your shell script.

Please quote your strings properly, single quotes don't hurt.
In your code `.["input-decimal"]` is interpreted as a shell glob.

Example:
```sh
touch .i
echo '{"i": "hello bug"}' | jq -r .[\"input-decimal\"]
# prints hello bug
```

**Comment by brother** [comment]
> It is completely valid and refers to a bug in your shell script.

We are still guessing though as the issue description is not clear. To me this report is invalid and should be closed.

**Comment by xubuntu4iran** [comment]
Description added to first post, @sideeffect42 how to edit my script if it is not a bug

**Comment by xubuntu4iran** [comment]
> Please do not ignore this shellcheck message. It is completely valid and refers to a bug in your shell script.
> 
> Please quote your strings properly, single quotes don't hurt. In your code `.["input-decimal"]` is interpreted as a shell glob.
> 
> Example:
> 
> touch .i
> echo '{"i": "hello bug"}' | jq -r .[\"input-decimal\"]
> # prints hello bug

for me , it printed null
 echo '{"i": "hello bug"}' | jq -r .[\"input-decimal\"]
null

**Comment by sideeffect42** [comment]
> Description added to first post, [@sideeffect42](https://github.com/sideeffect42) how to edit my script if it is not a bug

@xubuntu4iran: like I had written already, you have to quote your JQ code:
```sh
comp=$(sed -n "${j}p" "${filename}" | jq -r '.["input-decimal"]')
```

> > Example:
> > touch .i
> > echo '{"i": "hello bug"}' | jq -r .["input-decimal"]
> 
> for me , it printed null echo '{"i": "hello bug"}' | jq -r .["input-decimal"] null

You have to run the complete example I posted.
First:
`touch .i`
And then:
`echo '{"i": "hello bug"}' | jq -r .[\"input-decimal\"]`

Or run just `touch .i` and then your code and you will see it misbehave.
